protocols/imap/src/main/java/org/apache/james/imap/decode/parser/SetACLCommandParser.java

```diff
@@ -43,10 +48,29 @@ public SetACLCommandParser(StatusResponseFactory statusResponseFactory) {
 
     @Override
     protected ImapMessage decode(ImapRequestLineReader request, Tag tag, ImapSession session) throws DecodingException {
-        final String mailboxName = request.mailbox();
-        final String identifier = request.astring();
-        final String rights = request.astring();
-        request.eol();
-        return new SetACLRequest(tag, mailboxName, identifier, rights);
+        try {
+            final SetACLRequest.MailboxName mailboxName = new SetACLRequest.MailboxName(request.mailbox());
+            final Username identifier = Username.of(request.astring());
+            final String editModeAndRights = request.astring();
+            request.eol();
+
+            MailboxACL.EditMode editMode = MailboxACL.EditMode.REPLACE;
+            if (StringUtils.isNotEmpty(editModeAndRights)) {
+                switch (editModeAndRights.charAt(0)) {
```

Comment:
We are missing a default branch in this switch. Because of the untested behavious coupled with the implicit default code path, mistakes are made downstreams and bug introduced.

Let's be specific and exhaustively describe the behaviour.